### PR TITLE
PLAT-4839: [Unit Test] Unit tests for helpdesk-service-client module

### PR DESCRIPTION
**Pull request by ardsilva**

### CODE REVIEW

File: helpdesk-ai/src/test/java/org/symphonyoss/symphony/bots/ai/impl/AiResponderImplTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package org.symphonyoss.symphony.bots.ai.impl;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class AiResponderImplTest {
```
Comment (robson-sanchez) [inline helpdesk-ai/src/test/java/org/symphonyoss/symphony/bots/ai/impl/AiResponderImplTest.java]: Please remove this class. It's out of scope for this PR.

File: helpdesk-service/helpdesk-service-client/pom.xml [hunk 2]
```diff
@@ -54,6 +54,16 @@
...
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-all</artifactId>
+            <version>2.0.2-beta</version>
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/pom.xml]: Please change it to 1.9.5 version and choose the test scope.

File: helpdesk-service/helpdesk-service-client/pom.xml [hunk 3]
```diff
@@ -54,6 +54,16 @@
             <artifactId>migbase64</artifactId>
             <version>2.2</version>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/pom.xml]: I don't think it's required once you have imported mockito-all as a dependency.

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  private static final String MOCK_MAKERCHECKER_ID = "MOCK_MAKERCHECKER_ID";
+  private static final Long MOCK_MAKER_ID = 123L;
+  private static final String DISPLAY_NAME = "DISPLAY_NAME";
+  private static final Long MOCK_USER_ID = 456L;
+  private static final String ATTACHMENT_ID = "ATTACHMENT_ID";
+  private static final String ATTACHMENT_NAME = "ATTACHMENT_NAME";
+  private static final String MESSAGE_ID = "MESSAGE_ID";
+  private static final Long TIMESTAMP = 1L;
+
+  private final String groupId = "GROUP_ID";
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: It should be a constant field.

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+  @Before
+  public void setUp() throws Exception {
+    Configuration.setDefaultApiClient(apiClient);
+    makercheckerClient = new MakercheckerClient(groupId, TICKET_SERVICE_URL);
+  }
+
+  @Test
+  public void getMakerchecker() throws ApiException {
+    doReturn(makercheckerMock()).when(apiClient)
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java:53]: Please try to be more specific in the mock parameters. At least define the real endpoint path (1st parameter) to make sure you are calling the correct API.

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        .invokeAPI(any(), eq("GET"), any(), any(), any(), any(), any(), any(), any(), any());
+
+    doReturn(MOCK_MAKERCHECKER_ID).when(apiClient).escapeString(any());
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: doReturn(MOCK_MAKERCHECKER_ID).when(apiClient).escapeString(MOCK_MAKERCHECKER_ID);

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+    assertNotNull(makerchecker);
+    assertEquals(makercheckerMock(), makerchecker);
+
+  }
+
+  @Test(expected = HelpDeskApiException.class)
+  public void getMakercheckerWithNullId() throws ApiException {
+
+    Makerchecker makerchecker = makercheckerClient.getMakerchecker(null);
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: You should test the error message.

```
try {
  Makerchecker makerchecker = makercheckerClient.getMakerchecker(null);
  fail();
} catch(HelpDeskApiException e) {
  assertEquals("Missing the required parameter 'id' when calling getMakerchecker", e.getMessage());
}
```

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+  }
+
+  @Test
+  public void createMakerchecker() throws ApiException {
+
+    doReturn(makercheckerMock()).when(apiClient)
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java:80]: Please try to be more specific in the mock parameters. At least define the real endpoint path (1st parameter) to make sure you are calling the correct API.

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+        .invokeAPI(any(), eq("POST"), any(), any(), any(), any(), any(), any(), any(), any());
+
+    Makerchecker makerchecker =
+        makercheckerClient.createMakerchecker(MOCK_MAKERCHECKER_ID, MOCK_MAKER_ID,
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: You missed testing the error scenario.

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+            MOCK_SERVICE_STREAM_ID, ATTACHMENT_ID, ATTACHMENT_NAME, MESSAGE_ID, TIMESTAMP, null);
+
+    assertNotNull(makerchecker);
+    assertEquals(makercheckerMock(), makerchecker);
+  }
+
+  @Test
+  public void updateMakerchecker() throws ApiException {
+
+    doReturn(MOCK_MAKERCHECKER_ID).when(apiClient).escapeString(any());
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: Please avoid any() when it's possible.

```
doReturn(MOCK_MAKERCHECKER_ID).when(apiClient).escapeString(MOCK_MAKERCHECKER_ID);
```

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+    Makerchecker makerchecker = makercheckerMock();
+    makerchecker.setChecker(null);
+
+    doReturn(makerchecker).when(apiClient).invokeAPI(any(),eq("PUT"),any(),any(),any(),any(),any(),any(),any(),any());
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: Please try to be more specific in the mock parameters. At least define the real endpoint path (1st parameter) and the makerchecker object to make sure you are calling the correct API with the proper content.

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+    assertNotNull(updatedMakerchecker);
+    assertEquals(makerchecker,updatedMakerchecker);
+
+  }
+
+  @Test(expected = HelpDeskApiException.class)
+  public void updateMakercheckerWithError() throws ApiException {
+
+    Makerchecker updatedMakerchecker = makercheckerClient.updateMakerchecker(new Makerchecker());
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: You should test the error message.

```
try {
  Makerchecker makerchecker = makercheckerClient.updateMakerchecker(new Makerchecker());
  fail();
} catch(HelpDeskApiException e) {
  assertEquals("Missing the required parameter 'id' when calling updateMakerchecker", e.getMessage());
}
```

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/ticket/client/TicketClientTest.java [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+    TicketSearchResponse response = new TicketSearchResponse();
+    response.add(ticket);
+
+    doReturn(response)
+        .when(apiClient)
+        .invokeAPI(any(), eq("GET"), any(), any(), any(), any(), any(), any(), any(), any());
+
+    ticket =  ticketClient.getUnresolvedTicketByClientStreamId(MOCK_CLIENT_STREAM_ID);
+
+    assertEquals(null,ticket);
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/ticket/client/TicketClientTest.java]: Use assertNull

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/ticket/client/TicketClientTest.java [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+
+    doReturn(TICKET_ID).when(apiClient).escapeString(TICKET_ID);
+
+    doReturn(ticket).when(apiClient)
+        .invokeAPI(any(), eq("PUT"), any(), any(), any(), any(), any(), any(), any(), any());
+
+    Ticket updatedTicket = ticketClient.updateTicket(ticket);
+
+    assertNotNull(updatedTicket);
+    assertFalse(getMockTicket().equals(updatedTicket));
```
Comment (robson-sanchez) [inline helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/ticket/client/TicketClientTest.java]: Weird assertion.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

File: helpdesk-ai/src/test/java/org/symphonyoss/symphony/bots/ai/impl/AiResponderImplTest.java (hunk 1 above)
Comment (ardsilva) [reply helpdesk-ai/src/test/java/org/symphonyoss/symphony/bots/ai/impl/AiResponderImplTest.java]: removed!

File: helpdesk-service/helpdesk-service-client/pom.xml (hunk 2 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/pom.xml]: ok

File: helpdesk-service/helpdesk-service-client/pom.xml (hunk 3 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/pom.xml]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java (hunk 4 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java (hunk 6 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java (hunk 5 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java:53]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java (hunk 7 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java (hunk 8 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java:80]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java (hunk 9 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java (hunk 10 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java (hunk 11 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java (hunk 12 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/makerchecker/client/MakercheckerClientTest.java]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/ticket/client/TicketClientTest.java (hunk 13 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/ticket/client/TicketClientTest.java]: ok

File: helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/ticket/client/TicketClientTest.java (hunk 14 above)
Comment (ardsilva) [reply helpdesk-service/helpdesk-service-client/src/test/org/symphonyoss/symphony/bots/helpdesk/service/ticket/client/TicketClientTest.java]: ok

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**